packages/components/src/custom-select-control-v2/legacy-component/index.tsx

```diff
@@ -78,7 +78,14 @@ function CustomSelectControl( props: LegacyCustomSelectProps ) {
 					value={ name }
 					children={ __experimentalHint ? withHint : name }
 					style={ style }
-					className={ className }
+					className={ clsx(
+						// Legacy classnames
+						'components-custom-select-control__item',
+						className,
+						{
+							'has-hint': !! __experimentalHint,
```

Comment:
Nit: should be fine without the type casting, right?

Reply:
Should be!  I removed the `!!` notation (even if for now that code [will stay commented out](https://github.com/WordPress/gutenberg/pull/62848#discussion_r1662682481))